### DISCUSSION TITLE
Installation

### DISCUSSION THREAD

**Discussion by MPatriot** [oqtane/oqtane.framework#1990] [General]
Forgive me for another long diatribe...

When I first downloaded and extracted the framework,, other than not having it extract into a double folder of "oqtane.framework-master", I just extracted it as is and never thought much of it.  First I learned how to build a development module and most everything seemed to work fine so when I started learning how to build themes I noticed they weren't being published with everything else, like the module was and I had to manually publish the theme via ftp.  I don't think I understood what it meant that the theme(s) had created and published to "oqtane.framework" or at least oqtane.theme.bootswatch did when I tried that out.  Also, when trying to manually make a module in my development module, those weren't showing up either.
Finally I decided to download the example DNF module I had been watching on YouTube so I could physically examine what was going on, I noticed the instructions said to install it at the same directory level as the framework I finally noticed that it was creating the "oqtane.framework" directory next to my "oqtane.framework-master" directory. I thought that was weird so I deleted that directory and compiled it again...then it hit me....doh!  It was literally expecting a directory named "oqtane.framework"  At first I was thinking How come the module I created wasn't doing that?  But after renaming the framework directory it compiled into that fine.  So, in the other directory I had been using to make my DNF like trial, I renamed the oqtane folder to "oqtane.framework" and then I saw that my module created the "oqtane.framework-master" directory and outputed to that!  Then I got it, somehow my development module knew where it hade been created from but the themes didn't and also the DNF module because it wasn't created from a directory named "oqtane.framework-master"
 
Sigh, that's when I decided to start over, again, rather than trying to figure out how to clean that mess up.   The only thing it really affected was a few jobs I had created (and lot of head scratching and time) but I had made a copy of the whole mess and was able to reuse the context / repository / code files I had already made.

I don't recall anything about it mattering what the name of the unzipped directory was but now knowing this, its huge and themes publish with everything fine!  Just thought others who might not know this might be glad to hear it before going through the same thing.

Cheers!
-Mark

**Comment by sbwalker** [comment]
You can certainly move projects around on your system however if you do so, you need to be sure to update the paths in the debug and release CMD files in the Package project in your module or theme solution.

**Reply by ghhv** [reply]
this got me for a long time too...
